public/app/plugins/datasource/loki/components/useLokiForceRefresh.ts

```diff
@@ -0,0 +1,27 @@
+import { useState, useEffect } from 'react';
+import { DatasourceStatus } from '@grafana/ui/src/types/plugin';
+
+export const useLokiForceRefresh = (
+  datasourceStatus: DatasourceStatus,
+  refreshLabels: (forceRefresh?: boolean) => void,
+  initialDatasourceStatus?: DatasourceStatus
+) => {
+  const [prevDatasourceStatus, setPrevDatasourceStatus] = useState(
```

Comment:
Don't think you need that state. Since the effect below is performed on `datasourceStatus` argument change only, then if the status changed to `connected` then you have all the information to perform labels refresh.

Reply:
Will it not refresh the first time then?